app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/DummyCategory.kt

```diff
@@ -0,0 +1,21 @@
+package com.github.HumanLearning2021.HumanLearningApp.model
+
+
+/**
+ * Class representing a dummy implementation of the category interface
+ *
+ * @param name the name of the category (case-insensitive)
+ */
+
+class DummyCategory(name: String):Category {
+    override val name = name.toLowerCase()
+
+    override fun equals(other: Any?): Boolean {
+        return other is Category && other.name == name
+    }
+
+    override fun hashCode(): Int {
+        return 17 + 31 * name.hashCode()
+    }
```

Comment:
Same: wouldn't a data class do the trick?

Reply:
Here it's easier not to, as I convert the name argument to lower case before making it a property, to have case-insensitivity between categories